src/components/project/import/useProjectImport.ts

```diff
@@ -0,0 +1,25 @@
+import {GroupAnalysisStoreImport} from "@/store/new/GroupAnalysisStore";
+import JSZip from "jszip";
+
+export default function useProjectImport() {
+    const process = async (file: File): Promise<GroupAnalysisStoreImport> => {
+        const zipper = await JSZip.loadAsync(file);
+
+        const metadata = JSON.parse(await zipper.file("metadata.json")?.async("string"));
+
+        const buffers = zipper.folder("buffers");
+
+        for (const group of metadata.groups) {
+            for (const analysis of group.analyses) {
+                analysis.indexBuffer = await buffers.file(`${analysis.id}.index`)?.async("arraybuffer") || undefined;
+                analysis.dataBuffer = await buffers.file(`${analysis.id}.data`)?.async("arraybuffer") || undefined;
+            }
+        }
+
+        return metadata;
+    };
+
+    return {
+        process
+    };
+}
```

Comment:
I don't think there's a version number of the current file structure stored in the exports anywhere? We will need this in the future if we ever add new features to Unipept which might need changes to the exported data.

If a user uploads a project that was created one year ago, it should still work. Using the file structure version, we can correctly convert older projects to a newer format then.

Reply:
I added the version of the current version to the export. This value is currently not used on import, but is available when we introduce future breaking changes